core/src/com/ericc/the/game/systems/realtime/RenderSystem.java

```diff
@@ -52,18 +57,21 @@ public void update(float deltaTime) {
         Gdx.gl.glClearColor(.145f, .075f, .102f, 1);
         Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT);
 
-        batch.setProjectionMatrix(viewport.getCamera().combined);
-        batch.setBlendFunction(GL20.GL_SRC_ALPHA, GL20.GL_ONE_MINUS_SRC_ALPHA);
-        batch.begin();
-
         // Compute the visible map area.
-        Vector2 topLeft = viewport.unproject(new Vector2(0, 0));
-        int top = clamp(0, (int) topLeft.y, map.height() - 1);
-        int left = clamp(0, (int) topLeft.x, map.width() - 1);
+        sbh.update();
```

Comment:
czy ten sbh to nie mógłby być znowu osobny system? Bo skoro on się updatuje co krok, to czemu wołać mu update tutaj? Niech sam się updatuje osobno x) to wygląda na niepotrzebny coupling